File: Backend/eLibrary/src/main/java/com/diploma/elibrary/service/interfaces/AccountService.java
package com.diploma.elibrary.service.interfaces;

import com.diploma.elibrary.model.Account;

import java.util.List;

public interface AccountService {
    boolean emailExists(String email);
    Account findByEmail(String email);
    Account updatePassword(Long id, String new_password);
    Account updateAccount(Long id, Account accountDetails);
    Account createAccount(Account account);
    List<Account> getUsers();

    Account updateAccountFullName(Long id, Account accountDetails);
}

File: Frontend/eLibrary-frontend/src/app/services/account.service.ts
import { Injectable } from '@angular/core';
import { ApiService } from './api.service';
import { AuthService } from './auth.service';
import { firstValueFrom } from 'rxjs';
import { Account } from '../entities/account';

@Injectable({
  providedIn: 'root'
})
export class AccountService {

  constructor(private apiService: ApiService, private authService: AuthService) {}
  
  registerAccount(account : Account) {
    return this.apiService.createAccount(account);
  }
  emailExists(email : string) {
    return firstValueFrom(this.apiService.emailExists(email));
  }
  getAccount(username : string) {
    return firstValueFrom(this.apiService.getAccount(username));
  }

  getUsers(){
    return this.apiService.getUsers();
  }

  updateAccount(account: Account) {
    return this.apiService.updateAccount(account);
  }
}
